src/Agent.Listener/Configuration/WindowsServiceControlManager.cs

```diff
@@ -11,77 +12,371 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+                    if (result == 0)
+                    {
+                        result = Marshal.GetLastWin32Error();
+                        Trace.Error(StringUtil.Format("Could not delete service, result: {0}", result));
+                        throw new InvalidOperationException(StringUtil.Loc("CouldNotRemoveService", serviceName));
+                    }
+
+                    Trace.Info("successfully removed the service");
+                }
+                finally
+                {
+                    CloseServiceHandle(serviceHndl);
+                }
+            }
+            finally
+            {
+                CloseServiceHandle(scmHndl);
+            }
         }
 
-        public bool Unconfigure(string serviceName)
+        protected virtual void InstallService(string serviceName, string serviceDisplayName, string logonAccount, string logonPassword)
         {
-            return true;
+            Trace.Entering();
+
+            string agentServiceExecutable = Path.Combine(IOUtil.GetBinPath(), WindowsServiceControllerName);
+            IntPtr scmHndl = OpenSCManager(null, null, ServiceManagerRights.AllAccess);
+
+            if (scmHndl.ToInt64() <= 0)
+            {
+                throw new Exception("Failed to Open Service Control Manager");
+            }
+
+            try
+            {
+                Trace.Verbose(StringUtil.Format("Opened SCManager. Trying to create service {0}", serviceName));
+
+                IntPtr serviceHndl = CreateService(
+                                        scmHndl,
+                                        serviceName,
+                                        serviceDisplayName,
+                                        ServiceRights.QueryStatus | ServiceRights.Start,
+                                        SERVICE_WIN32_OWN_PROCESS,
+                                        ServiceBootFlag.AutoStart,
+                                        ServiceError.Normal,
+                                        agentServiceExecutable,
+                                        null,
+                                        IntPtr.Zero,
+                                        null,
+                                        logonAccount,
+                                        logonPassword);
+
+                if (serviceHndl.ToInt64() <= 0)
+                {
+                    throw new InvalidOperationException(StringUtil.Loc("OperationFailed", nameof(CreateService), GetLastError()));
+                }
+
+                _term.WriteLine(StringUtil.Loc("ServiceConfigured", serviceName));
+                CloseServiceHandle(serviceHndl);
+            }
+            finally
+            {
+                CloseServiceHandle(scmHndl);
+            }
+        }
+
+        public override void StopService(string serviceName)
+        {
+            Trace.Entering();
+            try
+            {
+                var service = TryGetServiceController(serviceName);
```

Comment:
What type is `var`? It's not clear from the statement. Per style guidelines use the type name when it isn't clear.